pkg/backingstore/reconciler.go

```diff
@@ -350,26 +350,45 @@ func (r *Reconciler) ReadSystemInfo() error {
 	}
 	r.SystemInfo = &systemInfo
 
+	// Check if pool exists
+	for i := range r.SystemInfo.Pools {
+		p := &r.SystemInfo.Pools[i]
+		if p.Name == r.BackingStore.Name {
+			r.PoolInfo = p
+			break
+		}
+	}
+
+	pool := r.PoolInfo
+	if r.BackingStore.Spec.Type == nbv1.StoreTypePVPool {
+		if pool != nil && pool.ResourceType != "HOSTS" {
+			r.Logger.Warnf("BackingStore %q w/existing pool %+v has unexpected resource type %+v", r.BackingStore.Name, pool, pool.ResourceType)
```

Comment:
**[IMPORTANT]**
That's a good thing to check but I think we want to take a bit more drastic measures.
I suggest to return a persistent error so that this backing store will get to a rejected state.
Also this should probably be checked in the same manner for the opposite situation - that is when `r.BackingStore.Spec.Type != nbv1.StoreTypePVPool && pool != nil && pool.ResourceType != "CLOUD"`